src/currentprocess/filesource.rs

```diff
@@ -79,71 +79,38 @@ impl StdinSource for super::TestProcess {
     }
 }
 
-// -------------- stdout -------------------------------
-
-pub trait Isatty {
-    fn isatty(&self) -> bool;
-}
-
-/// Stand-in for std::io::StdoutLock
+/// Allows you to write to a terminal or other output stream.
 pub trait WriterLock: Write {}
 
-/// Stand-in for std::io::Stdout
-pub trait Writer: Write + Isatty + Send {
+/// Holds a terminal or other output stream.
+pub trait Writer: Send + Sync {
+    /// Locks the terminal so that you can write to it.
     fn lock(&self) -> Box<dyn WriterLock + '_>;
 }
 
-/// Stand-in for std::io::stdout
+// -------------- stdout -------------------------------
+
 pub trait StdoutSource {
-    fn stdout(&self) -> Box<dyn Writer>;
+    fn stdout(&self) -> Box<dyn Terminal>;
```

Comment:
This should still return Writer in my opinion. Can you help me understand why it needs to return Terminal?

Reply:
As you said in https://github.com/rust-lang/rustup/pull/3351#issuecomment-1556025976, we cannot abstract it with the writer.